src/main/webapp/app/quiz/participate/quiz.component.ts

```diff
@@ -824,48 +824,85 @@ export class QuizComponent implements OnInit, OnDestroy {
         }
     }
 
+    /**
+     * Checks if the student has interacted with each question of the quiz
+     * For a Multiple Choice Questions it checks if an answer option was selected
+     * For a Drag and Drop Questions it checks if at least one mapping has been made
+     * For a Short Answer Questions it checks if at least one field has been clicked in 
+     * Goes trough selectedAnswerOptions and dragAndDropMappings
+     */
+    checkForAnsweredQuestions(): boolean {
```

Comment:
The method name is not good. What does check mean? If you return a boolean you should ask a question in the method name, e.g. areAllQuestionsAnswered()
Please also use proper JavaDoc style comments and make sure to describe exactly what the return type means and when it will be true and when it will be false